zjit/src/hir/opt_tests.rs

```diff
@@ -6183,6 +6290,55 @@ mod hir_opt_tests {
         ");
     }
 
+    #[test]
+    fn test_dont_optimize_when_passing_too_many_args() {
+        eval(r#"
+            public def foo(lead, opt=raise) = opt
+            def test = 0.foo(3, 3, 3)
```

Comment:
why 0.foo vs just foo?

Reply:
o_O

How does it get `foo` defined on `0` though? I thought that modified the top-self?

Reply:
`def` at top level defines the method on `Object`, not on top-self.